Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -97,6 +98,36 @@ private List<Token> NumberWithMonth(string text)
                 }
             }
             return ret;
-        }
+        }
+
+        private List<Token> DurationWithBeforeAndAfter(string text)
+        {
+            var ret = new List<Token>();
+            var duration_er = config.DurationExtractor.Extract(text);
+            foreach (var er in duration_er)
+            {
+                var match = config.NonDateUnitRegex.Match(er.Text);
+                if (match.Success)
+                {
+                    continue;
+                }
+                var pos = (int)er.Start + (int)er.Length;
+                if (pos < text.Length)
+                {
+                    var tmp = text.Substring(pos + 1);
+                    if (tmp.StartsWith("ago"))

```

Comment:
BaseDateExtractor should not directly reference language-specific terms "ago" and "later".